config/database.yml

```diff
@@ -24,22 +22,49 @@ defaults: &defaults
     statement_timeout: <%= IdentityConfig.store.database_statement_timeout %>
 
 development:
-  <<: *defaults
+  primary:
+    <<: *defaults
+  primary_replica:
+    <<: *defaults
+    replica: true
 
 test:
-  <<: *defaults
-  pool: 10
-  checkout_timeout: 10
-  database: <%= ENV['POSTGRES_DB'] || "upaya_test#{ENV['TEST_ENV_NUMBER']}" %>
-  user: <%= ENV['POSTGRES_USER'] %>
-  password: <%= ENV['POSTGRES_PASSWORD'] %>
+  primary: &test
+    <<: *defaults
+    pool: 10
+    checkout_timeout: 10
+    database: <%= ENV['POSTGRES_DB'] || "upaya_test#{ENV['TEST_ENV_NUMBER']}" %>
+    user: <%= ENV['POSTGRES_USER'] %>
+    password: <%= ENV['POSTGRES_PASSWORD'] %>
+  primary_replica:
+    <<: *test
+    replica: true
+
+<%
+  pool = if Identity::Hostdata.instance_role == 'worker'
+    IdentityConfig.store.good_job_max_threads * IdentityConfig.store.database_pool_idp
+  else
+    IdentityConfig.store.database_pool_idp
+  end
```

Comment:
cc @mitchellhenke @jgrevich this was how we wanted to scale threads for workers? or did we want `max_threads + pool_idp` (pool is 5 here)

Reply:
Jobs will most likely using at least one connection each, so adding together makes sense to me to give a little bit of breathing room. I'm not sure what the max total connections is though.

Reply:
Updated to add in 216b3e988ab2fd4de8cd9c921327796c54868e9c